### PR TITLE
refactored BookRepository, added delete, search methods to bookService

### PR DESCRIPTION
**Pull request by mmosii**

### CODE REVIEW

File: src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -35,4 +39,30 @@ public List<BookDto> findAll() {
...
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can't find book with id " + id);
+        }
+        Book book = bookMapper.toBook(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
```
Comment (teract10s) [inline src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java]: I'm not sure if this will work if someone doesn't give every field

**Review by teract10s — APPROVED** [review]
Good job. I left one minor comment

File: src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -35,4 +39,30 @@ public List<BookDto> findAll() {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("Can't find book with id " + id);
+        }
```
Comment (liliia-ponomarenko) [inline src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java]: It can be a private method

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! But Let’s fix one comment before merging ;)

**Pull request merged**